engine/src/main/java/org/terasology/engine/world/chunks/remoteChunkProvider/RemoteChunkProvider.java

```diff
@@ -61,11 +59,12 @@
     private final ChunkProcessingPipeline loadingPipeline;
     private EntityRef worldEntity = EntityRef.NULL;
     private ChunkReadyListener listener;
+    private ReceivedChunkInitialProvider initialProvider;
 
     public RemoteChunkProvider(BlockManager blockManager, LocalPlayer localPlayer) {
         this.blockManager = blockManager;
-        loadingPipeline = new ChunkProcessingPipeline(this::getChunk,
-            new LocalPlayerRelativeChunkComparator(localPlayer));
+        initialProvider = new ReceivedChunkInitialProvider(new LocalPlayerRelativeChunkComparator(localPlayer));
+        loadingPipeline = new ChunkProcessingPipeline(this::getChunk, initialProvider);
```

Comment:
How does this even work? Do we have multiple `RemoteChunkProvider`s, one per player? Or how are other players (clients) taken into account?

Reply:
There's one instance of `RemoteChunkProvider` on each client, which receives chunks from the server and submits them to the pipeline. That's in contrast to the `LocalChunkProvider`, which runs on the server and generates chunks itself.